src/mlio/default_instance_reader.cxx

```diff
@@ -158,20 +161,58 @@ default_instance_reader::handle_nested_errors()
 std::optional<memory_slice>
 default_instance_reader::read_record_payload()
 {
+    std::string split_record_err_str = "Split record is not formatted with the correct cflag.";
+    if (this->corrupt_split_record_) {
+        throw corrupt_record_error(split_record_err_str); 
+    }
     std::optional<record> rec = read_record();
     if (rec == std::nullopt) {
         return {};
     }
 
-    if (rec->kind() != record_kind::complete) {
-        throw not_supported_error{"Split records are not supported yet!"};
-    }
+    if (rec->kind() == record_kind::complete) {
+        num_bytes_read_ += rec->size();
+        store_record_idx_++;
+
+        return std::move(*rec).payload();
+    } else {
+        if (rec->kind() == record_kind::begin) {
+            std::vector<std::optional<record>> split_records;
+            split_records.push_back(rec);
+            std::size_t total_record_size = 0;
+            total_record_size += rec->size();
+
+            std::optional<record> next_rec = read_record();
+            while(next_rec->kind() == record_kind::middle) {
+                split_records.push_back(next_rec);
+                total_record_size += next_rec->size();
+                next_rec = read_record();
+            }
+            
+            if (next_rec->kind() == record_kind::end) {
+                split_records.push_back(next_rec);
+                total_record_size += next_rec->size();
+            } else {
+                this->corrupt_split_record_ = true;
+                throw corrupt_record_error(split_record_err_str);
+            }
 
-    num_bytes_read_ += rec->size();
+            auto combined_blk = get_memory_allocator().allocate(total_record_size);
+            auto copied_so_far_blk = combined_blk->begin();
 
-    store_record_idx_++;
+            for (auto it = split_records.begin(); it < split_records.end(); it++) {
+                auto current_payload = std::move(*it)->payload();
+                copied_so_far_blk = std::copy(current_payload.begin(), current_payload.end(), copied_so_far_blk);
+            }
 
-    return std::move(*rec).payload();
+            num_bytes_read_ += total_record_size;
+            store_record_idx_++;
+            return std::move(combined_blk); 
```

Comment:
The compiler uses return value optimization anyways. You should avoid calling `std::move` here.

Reply:
The compiler was giving me a warning to add this in case or version differences.

Reply:
```
../../src/mlio/default_instance_reader.cxx:215:20: note: call 'std::move' explicitly to avoid copying on older compilers 
```